File: src/seminar05_hw/Core/Models/Worker.java
package seminar05_hw.Core.Models;

public class Worker {
    String fullName;
    String dateOfBirth;
    Division division;
    Position position;
    Integer salary;


    private Worker(WorkerBuilder workerBuilder) {
        fullName = workerBuilder.fullName;
        dateOfBirth = workerBuilder.dateOfBirth;
        division = workerBuilder.division;
        salary = workerBuilder.salary;
        position = workerBuilder.position;
    }

    public String extendedInfo() {
        return "ФИО: " + fullName + "\n" +
                "Дата рождения: " + dateOfBirth + "\n" +
                "Подразделение: " + division.getUnit() + "\n" +
                "Должность: " + position.getUnit() + "\n" +
                "Оклад: " + salary + "\n";
    }

    public String info() {
        return fullName + "\n" +
                dateOfBirth + "\n" +
                division.getUnit() + "\n" +
                position.getUnit() + "\n" +
                salary + "\n";
    }

    public String getFullName() {
        return this.fullName;
    }


    public static class WorkerBuilder {
        private String fullName;
        private String dateOfBirth;
        private Division division;
        private Position position;
        private Integer salary;


        public WorkerBuilder(String fullName) {
            this.fullName = fullName;
            this.dateOfBirth = "Неизвестно";
            this.division = new Division();
            this.position = new Position();
            this.salary = 11000;
        }

        public WorkerBuilder setDateOfBirth(String dateOfBirth) {
            this.dateOfBirth = dateOfBirth;
            return this;
        }

        public WorkerBuilder setSalary(Integer salary) {
            this.salary = salary;
            return this;
        }

        public WorkerBuilder setDivision(String division) {
            this.division.setUnit(division);
            return this;
        }

        public WorkerBuilder setPosition(String position) {
            this.position.setUnit(position);
            return this;
        }

        public Worker build() {
            return new Worker(this);
        }

    }
}

File: src/seminar07_hw/models/Worker.py
from models.Division import Division
from models.Position import Position


class Worker_Builder:
    _full_name: str
    _date_of_birth: str = "Неизвестно"
    _division: Division
    _position: Position
    _salary: int

    def __init__(self, full_name: str):
        self._full_name = full_name
        self._date_of_birth = "Неизвестно"
        self._division = Division()
        self._position = Position()
        self._salary = 11000

    def set_full_name(self, full_name: str):
        self._full_name = full_name
        return self

    def set_date_of_birth(self, date_of_birth: str):
        self._date_of_birth = date_of_birth
        return self

    def set_division(self, division: str):
        self._division.set_division(division)
        return self

    def set_position(self, position: str):
        self._position.set_position(position)
        return self

    def set_salary(self, salary: int):
        self._salary = salary
        return self

    def build(self):
        return Worker(self)


class Worker:
    __full_name: str
    __date_of_birth: str
    __division: Division
    __position: Position
    __salary: int

    def __init__(self, worker_builder: Worker_Builder):
        self.__full_name = worker_builder._full_name
        self.__date_of_birth = worker_builder._date_of_birth
        self.__division = worker_builder._division
        self.__position = worker_builder._position
        self.__salary = worker_builder._salary

    def get_worker_list(self):
        return [self.get_full_name(),
                self.get_date_of_birth(),
                self.get_division(),
                self.get_position(),
                self.get_salary()]

    def get_full_name(self):
        return self.__full_name

    def get_date_of_birth(self):
        return self.__date_of_birth

    def get_division(self):
        return self.__division.get_division()

    def get_position(self):
        return self.__position.get_position()

    def get_salary(self):
        return self.__salary

    # def get_info:
    #
